integration-cli/docker_api_containers_test.go

```diff
@@ -256,28 +257,45 @@ func TestVolumesFromHasPriority(t *testing.T) {
 
 func TestGetContainerStats(t *testing.T) {
 	defer deleteAllContainers()
-	name := "statscontainer"
-
-	runCmd := exec.Command(dockerBinary, "run", "-d", "--name", name, "busybox", "top")
+	var (
+		wg     = &sync.WaitGroup{}
+		name   = "statscontainer"
+		runCmd = exec.Command(dockerBinary, "run", "-d", "--name", name, "busybox", "top")
+	)
 	out, _, err := runCommandWithOutput(runCmd)
 	if err != nil {
 		t.Fatalf("Error on container creation: %v, output: %q", err, out)
 	}
+	type b struct {
+		body []byte
+		err  error
+	}
+	bc := make(chan b, 1)
+	wg.Add(2)
+	go func() {
+		defer wg.Done()
+		body, err := sockRequest("GET", "/containers/"+name+"/stats", nil)
+		bc <- b{body, err}
+	}()
 	go func() {
+		defer wg.Done()
 		time.Sleep(4 * time.Second)
 		runCommand(exec.Command(dockerBinary, "kill", name))
 		runCommand(exec.Command(dockerBinary, "rm", name))
 	}()
-
-	body, err := sockRequest("GET", "/containers/"+name+"/stats", nil)
-	if err != nil {
-		t.Fatalf("GET containers/stats sockRequest failed: %v", err)
+	// collect the results from the stats stream
+	sr := <-bc
```

Comment:
This is exactly one of `wg` jobs. also, it seems that this channel not spawn until you remove container. So I think we don't need `wg`. We need select with timeout to check that client disconnects on container removal.